src/rpc/services/SpectateService.ts

```diff
@@ -0,0 +1,531 @@
+import { ARpcService, MainRpcService } from "@/rpc";
+import { ChannelId, Timestamp, Hash } from "@/types/types";
+import { StateSnapshot } from "@/models";
+import Clock from "@/Clock";
+import ATransport from "@/transport/ATransport";
+import {
+    MilestoneProofStruct,
+    StateProofStruct
+} from "@typechain-types/contracts/V1/types/ProofTypes";
+import { DisputeStruct } from "@typechain-types/contracts/V1/types/DisputeTypes";
+import { Codec, Type } from "@/utils";
+import { isEqual } from "lodash";
+
+export interface SnapshotPayload {
+    latestForkGenesisSnapshot: StateSnapshot;
+    disputeWindows?: DisputeStruct[][];
+    stateProof?: StateProofStruct;
```

Comment:
you also need milestoneSnapshots as used and required in the stateProof verification logic